tun: fix crash when ForceMTU is called after close

raggi opened the pull request:
Close closes the events channel, resulting in a panic from send on closed channel.

Reported-By: Brad Fitzpatrick <brad@tailscale.com>
Link: https://github.com/tailscale/tailscale/issues/9555

bradfitz commented:
> Want to also send this as a PR to wireguard/wireguard-go? (Include a Signed-off-by if you do.)

Or should I do that instead and then pull it down to here? Is that happening regularly/cleanly?

crawshaw commented:
Generally, if you can send a PR upstream first, please do.

It's time for us to rebase tailscale/wireguard-go on wireguard/wireguard-go. But for clean changes like this, it's fine to just commit it on top and delay the rebase.

raggi committed:
tun: fix crash when ForceMTU is called after close

Close closes the events channel, resulting in a panic from send on
closed channel.

Reported-By: Brad Fitzpatrick <brad@tailscale.com>
Link: https://github.com/tailscale/tailscale/issues/9555
Signed-off-by: James Tucker <james@tailscale.com>

raggi commented:
I have sent this upstream too, this is what I'd like to pick onto our working branch.
